File: backend_dart/script_delete_users.dart
import 'dart:io';
import 'package:backend_dart/config/env.dart';
import 'package:backend_dart/config/firebase.dart';

Future<void> main() async {
  print('Starting user wipe script using Backend Config...');
  env.load();
  
  await FirebaseConfig.initialize();
  
  if (!FirebaseConfig.isInitialized) {
    print('Failed to initialize Firebase! Exiting.');
    exit(1);
  }

  try {
    final firestore = FirebaseConfig.db!;
    final auth = FirebaseConfig.auth!;
    
    print('Fetching Firestore users...');
    final docs = await firestore.collection('users').get();
    print('Found ${docs.docs.length} users in Firestore.');
    for (final doc in docs.docs) {
      await doc.ref.delete();
      print('Deleted Firestore doc: ${doc.id}');
    }

    print('Fetching Firebase Auth users...');
    final listUsersResult = await auth.listUsers();
    final authUsers = listUsersResult.users;
    print('Found ${authUsers.length} users in Firebase Auth.');
    for (final user in authUsers) {
      await auth.deleteUser(user.uid);
      print('Deleted Auth user: ${user.uid} (${user.email})');
    }

    print('Successfully wiped all users from Firestore and Firebase Auth.');
    exit(0);
  } catch (e, stack) {
    print('Error during deletion: $e\n$stack');
    exit(1);
  }
}

File: backend_dart/script_delete_users.js
const admin = require('firebase-admin');
const dotenv = require('dotenv');

dotenv.config();

async function main() {
  console.log('Starting user wipe script (Node.js Admin SDK)...');

  const base64String = process.env.FIREBASE_SERVICE_ACCOUNT_BASE64;
  if (!base64String) {
    console.error('FIREBASE_SERVICE_ACCOUNT_BASE64 missing from environment.');
    process.exit(1);
  }

  try {
    const jsonString = Buffer.from(base64String, 'base64').toString('utf-8');
    const serviceAccount = JSON.parse(jsonString);

    // Clean private key just like Dart wrapper does
    if (serviceAccount.private_key) {
       serviceAccount.private_key = serviceAccount.private_key.replace(/\\n/g, '\n');
    }

    const app = admin.initializeApp({
      credential: admin.credential.cert(serviceAccount)
    });

    const firestore = admin.firestore(app);
    const auth = admin.auth(app);

    console.log('Fetching Firestore users...');
    const snapshot = await firestore.collection('users').get();
    console.log(`Found ${snapshot.docs.length} users in Firestore.`);
    
    // Batch delete Firestore users
    const batch = firestore.batch();
    snapshot.docs.forEach((doc) => {
      batch.delete(doc.ref);
      console.log(`Prepared deletion of Firestore doc: ${doc.id}`);
    });
    
    if (snapshot.docs.length > 0) {
      await batch.commit();
      console.log('Firestore batch delete committed successfully.');
    }

    // console.log('\Fetching Firebase Auth users...');
    // let listUsersResult = await auth.listUsers(1000);
    // let authUsers = listUsersResult.users;
    // console.log(`Found ${authUsers.length} users in Firebase Auth.`);
    // 
    // for (const user of authUsers) {
    //   await auth.deleteUser(user.uid);
    //   console.log(`Deleted Auth user: ${user.uid} (${user.email || 'No email'})`);
    // }

    console.log('\nSuccessfully wiped all users from Firestore.');
    process.exit(0);
  } catch (error) {
    console.error('Error during deletion:', error);
    process.exit(1);
  }
}

main();
